From b2511e1762c76af5df9738982b8ca67652e1f582 Mon Sep 17 00:00:00 2001
From: Jiping Ma <jiping.ma2@windriver.com>
Date: Mon, 17 Dec 2018 02:49:00 +0000
Subject: [PATCH] Yaffs2: The file mode is not correct after copy file to
 yaffs2 file system such as the command "cp -a".

yaffs_setxattr() did not modify inode related mode info with
YAFFS_XATTR enable.

Signed-off-by: Jiping Ma <jiping.ma2@windriver.com>
---
 fs/yaffs2/yaffs_vfs.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/fs/yaffs2/yaffs_vfs.c b/fs/yaffs2/yaffs_vfs.c
index b029b8d4276e..c4e6931c46d5 100644
--- a/fs/yaffs2/yaffs_vfs.c
+++ b/fs/yaffs2/yaffs_vfs.c
@@ -920,6 +920,7 @@ static int yaffs_setattr(struct dentry *dentry, struct iattr *attr)
 	return error;
 }
 
+#ifdef CONFIG_YAFFS_XATTR
 static int yaffs_setxattr(struct dentry *dentry, struct inode *inode,
 			  const char *name, const void *value, size_t size,
 			  int flags)
@@ -1003,6 +1004,7 @@ static int yaffs_removexattr(struct dentry *dentry, const char *name)
 
 	return error;
 }
+#endif
 
 static ssize_t yaffs_listxattr(struct dentry * dentry, char *buff, size_t size)
 {
@@ -2775,6 +2777,7 @@ static struct dentry *yaffs_make_root(struct inode *inode)
 }
 
 
+#ifdef CONFIG_YAFFS_XATTR
 static int yaffs_xattr_get(const struct xattr_handler *handler,
 			    struct dentry *dentry, struct inode *inode,
 			    const char *name, void *buff, size_t size)
@@ -2803,6 +2806,7 @@ static const struct xattr_handler *yaffs_xattr_handlers[] = {
 	&yaffs_xattr_handler,
 	NULL
 };
+#endif
 
 static struct super_block *yaffs_internal_read_super(int yaffs_version,
 						     struct super_block *sb,
@@ -2836,7 +2840,9 @@ static struct super_block *yaffs_internal_read_super(int yaffs_version,
 
 	sb->s_magic = YAFFS_MAGIC;
 	sb->s_op = &yaffs_super_ops;
+#ifdef CONFIG_YAFFS_XATTR
 	sb->s_xattr = yaffs_xattr_handlers;
+#endif
 	sb->s_flags |= MS_NOATIME;
 
 	read_only = ((sb->s_flags & MS_RDONLY) != 0);
-- 
2.18.1


